driver/ppm_fillers.c

```diff
@@ -5735,7 +5738,8 @@ int f_sys_dup_x(struct event_filler_arguments *args)
 	 * oldfd
 	 */
 	syscall_get_arguments_deprecated(args, 0, 1, &val);
-	res = val_to_ring(args, val, 0, false, 0);
+	fd = (int64_t) val;
```

Comment:
for example here the cast should be `int32_t` so probably is better to use a unique pattern everywhere to avoid issues